src/Language/Spectacle/Syntax/Modal.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiWayIf #-}
+{-# LANGUAGE TupleSections #-}
+
+module Language.Spectacle.Syntax.Modal
+  ( Modal (Modal),
+    Effect (Always, UpUntil),
+    always,
+    eventually,
+    upUntil,
+    testModalResult1,
+  )
+where
+
+import Control.Monad
+import Data.Coerce
+import Data.Void
+
+import Data.Functor.Loom
+import Language.Spectacle.Lang
+import Language.Spectacle.Syntax.Modal.Internal
+import Language.Spectacle.Syntax.Modal.Quoted
+import Language.Spectacle.Syntax.Logic
+import Language.Spectacle.Syntax.NonDet
+import Language.Spectacle.Syntax.Error
+import Language.Spectacle.Exception.RuntimeException
+import Data.Function
+
+-- -------------------------------------------------------------------------------------------------
+
+always :: Member Modal effs => Lang ctx effs Bool -> Lang ctx effs Bool
+always m = scope (Always m)
+
+eventually :: Member Modal effs => Lang ctx effs Bool -> Lang ctx effs Bool
```

Comment:
I think it would be great to provide an explanation in the documentation for this syntax that it is defined in terms of `upUntil` and perhaps explain why (or link to an explanation).